packages/create-cloudflare/src/workers.ts

```diff
@@ -1,188 +1,156 @@
+		newToml.prepend(
+			`compatibility_date = "${await getWorkerdCompatibilityDate()}"`
+		);
 	}
-	await gitCommit(ctx);
-	endSection("Dependencies Installed");
-	if (!preexisting) {
-		await offerToDeploy(ctx);
-		await runDeploy(ctx);
+
+	const nameRe = /^name\s*=.*/m;
+	if (wranglerToml.match(nameRe)) {
+		newToml.replace(nameRe, `name = "${ctx.project.name}"`);
+	} else {
+		newToml.prepend(`name = "${ctx.project.name}"`);
 	}
 
-	await printSummary(ctx);
+	writeWranglerToml(ctx, newToml.toString());
 };
 
-async function getTemplate(ctx: Context) {
-	const preexisting = ctx.args.type === "pre-existing";
-	const template = preexisting ? "hello-world" : ctx.args.type;
-	const path = resolve(
-		// eslint-disable-next-line no-restricted-globals
-		__dirname,
-		"..",
-		"templates",
-		template,
-		ctx.args.ts ? "ts" : "js"
-	);
+/**
+ * Installs the latest version of the `@cloudflare/workers-types` package
+ * and updates the .tsconfig file to use the latest entrypoint version.
+ */
+export async function installWorkersTypes(ctx: C3Context) {
+	if (!usesTypescript(ctx)) {
+		return;
+	}
 
-	return { preexisting, template, path };
+	await installPackages(["@cloudflare/workers-types"], {
+		dev: true,
+		startText: "Installing @cloudflare/workers-types",
+		doneText: `${brandColor("installed")} ${dim(`via ${npm}`)}`,
+	});
+	await addWorkersTypesToTsConfig(ctx);
 }
 
-async function copyFiles(ctx: Context) {
-	const { template, path: srcdir } = await getTemplate(ctx);
-	const destdir = ctx.project.path;
+export async function addWorkersTypesToTsConfig(ctx: C3Context) {
+	const tsconfigPath = join(ctx.project.path, "tsconfig.json");
+	if (!existsSync(tsconfigPath)) {
+		return;
+	}
 
-	// copy template files
-	updateStatus(`Copying files from "${template}" template`);
-	await cp(srcdir, destdir, { recursive: true });
+	const s = spinner();
+	s.start("Adding latest types to `tsconfig.json`");
 
-	// reverse renaming from build step
-	await rename(join(destdir, "__dot__gitignore"), join(destdir, ".gitignore"));
-}
+	const tsconfig = readFile(tsconfigPath);
+	const entrypointVersion = getLatestTypesEntrypoint(ctx);
+	if (entrypointVersion === null) {
+		s.stop(
+			`${brandColor(
+				"skipped"
+			)} couldn't find latest compatible version of @cloudflare/workers-types`
+		);
+		return;
+	}
 
-async function copyExistingWorkerFiles(ctx: Context) {
-	const { preexisting } = await getTemplate(ctx);
-
-	if (preexisting) {
-		await chooseAccount(ctx);
-
-		if (ctx.args.existingScript === undefined) {
-			ctx.args.existingScript = await processArgument<string>(
-				ctx.args,
-				"existingScript",
-				{
-					type: "text",
-					question:
-						"Please specify the name of the existing worker in this account?",
-					label: "worker",
-					defaultValue: ctx.project.name,
-				}
-			);
+	const typesEntrypoint = `@cloudflare/workers-types/${entrypointVersion}`;
+
+	let updated = tsconfig;
+
+	if (tsconfig.includes("@cloudflare/workers-types")) {
```

Comment:
Note that this will be true if the workers-types already references a specific entry point version.
For example if the code looks like:

```
  "types": ["@cloudflare/workers-types/2023-20-20"],
```

We will update it to

```
  "types": ["@cloudflare/workers-types/2023-20-20/2023-20-20"],
```